src/components/Composer.vue

```diff
@@ -1209,16 +1216,27 @@ export default {
 			await this.checkRecipientsKeys()
 		},
 		onNewToAddr(option) {
-			this.onNewAddr(option, this.selectTo)
+			this.onNewAddr(option, this.selectTo, 'to')
 		},
 		onNewCcAddr(option) {
-			this.onNewAddr(option, this.selectCc)
+			this.onNewAddr(option, this.selectCc, 'cc')
 		},
 		onNewBccAddr(option) {
-			this.onNewAddr(option, this.selectBcc)
-		},
-		onNewAddr(option, list) {
-			if (list.some((recipient) => recipient.email === option.email)) {
+			this.onNewAddr(option, this.selectBcc, 'bcc')
+		},
+		onNewAddr(option, list, type) {
+			if (
+				(option === null || option === undefined)
+				&& this.recipientSearchTerms[type] !== undefined
+				&& this.recipientSearchTerms[type] !== ''
+			) {
+				option = {}
+				option.email = this.recipientSearchTerms[type]
+				option.label = this.recipientSearchTerms[type]
+				this.recipientSearchTerms[type] = ''
+			}
```

Comment:
I'm not sure I understand this trick. Is this for when you leave the input when a suggestion is shown in the dropdown?

Reply:
Took me a while to remember what was going on here too 😅

---

So ```(option === null || option === undefined)``` is needed because that function is called in two cases:
1. You selected one of the suggested options from the dropdown
2. You click outside the dropdown, so the recipient is added anyway

If you were to click on the dropdown without this check it would fire twice, once for the selection of the dropdown (`@option:selecting`), and once for the click closing the dropdown (`@search:blur`).

---

```
&& this.recipientSearchTerms[type] !== undefined
&& this.recipientSearchTerms[type] !== ''
``` 
Just normal checks

---
```
option = {}
option.email = this.recipientSearchTerms[type]
option.label = this.recipientSearchTerms[type]
```
Overriding the `option` parameter with not the dropdown content but the string content of the search of the `NcSelect`

```
this.recipientSearchTerms[type] = ''
```
Resetting the string content of the search of the `NcSelect` seeing as that string has already been added as a recipient

Hope this clears things up @ChristophWurst :)